[FEATURE] Support OTX for OSX

JihwanEom opened the pull request:
## This PR includes
- Support OTX for OSX, **validated only with python3.8.16, Ventura 13.1**

### Limitations
- Raise `pickle.lock` error when num_workers > 0 in dataloader. (Caused by different multiprocessing policy between Ubuntu & OSX)
- Support CPU only, MPS will be supported in future.
 
### How to install OTX on my Mac
```
pip install torch==1.13.1 torchvision==0.14.1
pip install -e '.[full]'
```
OR 
```
pip install tox
tox devenv venv/otx -e pre-merge
source venv/otx/bin/activate
```

### TODOs
- [x] Pass all of integration tests locally
- [ ] Pass all of e2e tests locally
- [ ] Pass all of unit tests locally

### Test results (E2E)

#### Unit test
<img width="1251" alt="image" src="https://user-images.githubusercontent.com/72460430/226187399-efed84ac-3253-4db5-80a9-1ee7a4d6fa80.png">

#### Classification
<img width="1339" alt="image" src="https://user-images.githubusercontent.com/72460430/226191143-b48b4e19-6a48-4c38-b271-85ff93667ffd.png">

#### Detection
<img width="1353" alt="image" src="https://user-images.githubusercontent.com/72460430/226223716-9291d6db-caeb-4989-9b47-ae00e62b9934.png">

#### Segmentation
<img width="1120" alt="image" src="https://user-images.githubusercontent.com/72460430/226218783-13969251-a99c-43a9-9aff-d9cf59e7ce24.png">

<img width="1336" alt="image" src="https://user-images.githubusercontent.com/72460430/226218759-e6ddf512-202e-4699-ad91-263c86b0e36a.png">

JihwanEom committed:
Mac Support

eunwoosh reviewed (dismissed):
LGTM. But thread lock isn't pickled in latest python version. I think other problem is the reason of that.

JihwanEom commented:
This PR will be reopened after passing all the integration cli tests in local Mac device.

yunchu committed:
[CI] Updated daily workflow (#1904)

Updated daily workflow

- remove if statement to allow running on any branch by manually

JihwanEom committed:
Merge branch 'develop' of https://github.com/openvinotoolkit/training_extensions into jeom/mac-support

JihwanEom committed:
Handle integration test fails in Mac

JihwanEom committed:
Update documentation

JihwanEom committed:
Support MPS only for multi-class classification

JihwanEom committed:
Update documentation

JihwanEom committed:
Align to OSX

JihwanEom committed:
Fix documentation

JihwanEom committed:
Update tox.ini

JihwanEom committed:
Fix documentation

JihwanEom committed:
Disable MPS

JihwanEom committed:
Disable MPS

JihwanEom commented:
Close by #1924
